### ISSUE TITLE
Disable automatic message deletion in SimpleMessageListenerContainer

### ISSUE THREAD

**Issue by alainsahli** [spring-cloud/spring-cloud-aws#51]
Add a way to completely disable the deletion of messages by the SimpleMessageListenerContainer.

**Comment by leocwolter** [comment]
Any news here?

**Comment by alainsahli** [comment]
I'm going to introduce a way to define the deletion strategy. I see 4 deletion strategies:
* ALWAYS: always deletes the messages
* NEVER: never deletes the messages
* ONLY_SUCCESSFUL_MESSAGES: deletes only messages that were successfully processed by the handler method
* ALWAYS_EXCEPT_REDRIVE: messages are always deleted except when an exception occurs in the handler method and the queue has a redrive policy

**Comment by artembilan** [comment]
I'd say that this looks like `Manual Acknowledge`.
We recently introduced some `Acknowledgment ` abstraction for the `KafkaMessageListenerContainer`: https://github.com/spring-projects/spring-integration-kafka/blob/master/src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java#L68

It would be better to have here something similar, to allow end-user to control this `deleteMessage` stuff, e.g. via `ACKNOWLEDGMENT` message header in any downstream flow.

**Comment by ulisses79** [comment]
Hi, any news on that? It would be very useful enhancement.

**Comment by alainsahli** [comment]
Hi @ulisses79, you can set this field to `false` using Java config or XML config:

````java
@Bean
public SimpleMessageListenerContainerFactory simpleMessageListenerContainerFactory() {
    SimpleMessageListenerContainerFactory factory = new SimpleMessageListenerContainerFactory();
    factory.setDeleteMessageOnException(DELETE_MESSAGE_ON_EXCEPTION);
    
    return factory;
}
````

````xml
<aws-messaging:annotation-driven-queue-listener delete-message-on-exception="false" />
`````

**Comment by mobiletoly** [comment]
What if I don't want to throw an exception in order to suppress deletion of message? Why it is so hard to simply add this functionality??? This feature was requested long time ago.

**Comment by alainsahli** [comment]
Fixed by ae567dad0cf21ed729e432653356d7467c9b4eb6. Pay attention that with this commit `SimpleMessageListenerContainerFactory#setDeleteMessageOnException` was removed as the deletion policy is now defined on the listener method level.

If you want to disable the message deletion on a listener method use `SqsMessageDeletionPolicy.NEVER` as shown below:

```java
@SqsListener(value = "ManualDeletionQueue", deletionPolicy = SqsMessageDeletionPolicy.NEVER)		public void receive(String message, Acknowledgment acknowledgment)  {
    // this triggers the deletion of the message.
    acknowledgment.acknowledge();
}
```
